Makefile

```diff
@@ -291,8 +304,10 @@ controller-manager-image: build/controller-manager/Dockerfile $(BINDIR)/controll
 	rm -rf build/controller-manager/tmp
 
 # Push our Docker Images to a registry
+# note: .push-build-image should be the first dependency, because all others rely on the build
+# image
 ######################################
-push: user-broker-push controller-manager-push apiserver-push
+push: .push-build-image user-broker-push controller-manager-push apiserver-push
```

Comment:
I don't think pushing the build image should be lumped in with the rest of the pushes. They're two totally different actions.

`user-broker-push`, `controller-manager-push`, `apiserver-push`: these are pushing images of freshly compiled catalog binaries to your personal registry. `push-build-image`, on the other hand, is updating a shared image on a shared registry that is used during the build process. Adding `push-build-image` to this action makes it so you have to specifically name the user-broker, controller-manager, and apiserver actions if you just want to do your personal registry pushes.

If someone wants to push the build image, they can just explicitly call that command.